lib/spack/spack/build_systems/compiler.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import itertools
+import os
+import re
+from typing import Sequence
+
+import llnl.util.tty as tty
+from llnl.util.lang import classproperty
+
+import spack.compiler
+import spack.package_base
+
+
+class CompilerPackage(spack.package_base.PackageBase):
+    # metadata identifying this as a compiler for lmod
+    family = "compiler"
+
+    # Optional suffix regexes for searching for this type of compiler.
+    # Suffixes are used by some frameworks, e.g. macports uses an '-mp-X.Y'
+    # version suffix for gcc.
+    suffixes: Sequence[str] = [r"-.*"]
+
+    # Optional prefix regexes for searching for this compiler
+    prefixes: Sequence[str] = []
+
+    #: Compiler argument that produces version information
+    version_argument = "-dumpversion"
+
+    #: Return values to ignore when invoking the compiler to get its version
+    ignore_version_errors: Sequence[int] = ()
+
+    #: Regex used to extract version from compiler's output
+    version_regex = "(.*)"
+
+    #: Platform matcher for Platform objects supported by compiler
+    is_supported_on_platform = lambda x: True
```

Comment:
Let's not use lambdas at class level. We should define a method and make it clear if it is a static method (I assume `x` is the platform)

Reply:
I agree in principle, but I went and implemented it and it's really ugly.

The way I see it, there are 4 options:

1. `lambda x: True`
2. Define a static method in class scope
```
@staticmethod
def true():
    return True

is_supported_on_platform = true
```
3. Do the exact same thing with a method in module scope instead
4. `is_supported_on_platform = object`, because `object` is callable and always returns True.

I've implemented option 3, but I think I prefer the lambda over that (and that's what we had previously in the `spack.compiler.Compiler` class).